metacat-common-server/src/main/java/com/netflix/metacat/common/server/connectors/ConnectorFactoryDecorator.java

```diff
@@ -89,12 +114,18 @@ public ConnectorTableService getTableService() {
     public ConnectorPartitionService getPartitionService() {
         ConnectorPartitionService service = delegate.getPartitionService();
 
-        if (rateLimiterEnabled) {
-            log.info("Creating rate-limited connector partition services for connector-type: {}, "
-                         + "plugin-type: {}, catalog: {}, shard: {}",
+        if (validationEnabled) {
+            log.info("Creating validating connector partition services for connector-type: {}, "
+                    + "plugin-type: {}, catalog: {}, shard: {}",
                 connectorContext.getConnectorType(), connectorPlugin.getType(),
                 connectorContext.getCatalogName(), connectorContext.getCatalogShardName());
-            service = new ThrottlingConnectorPartitionService(service, rateLimiter);
+            service = new ValidatingConnectorPartitionService(
+                service,
+                rateLimiter,
+                rateLimiterEnabled,
+                authorization,
+                authorizationEnabled
```

Comment:
The authorization object should have the capability to determine its own enabled status, rather than having it flagged externally.